src/utils/HealthLogger.js

```diff
@@ -0,0 +1,170 @@
+     */
+    function getStoredHealthData() {
+        return PreferencesManager.getViewState(HEALTH_DATA_STATE_KEY);
+    }
+
+    /**
+     * Return the aggregate of all health data logged till now from all sources
+     * @returns {Object} Health Data aggregated till now
+     */
+    function getAggregatedHealthData() {
+        var healthData = getStoredHealthData();
+        $.extend(healthData, getPerformanceData());
+        return healthData;
+    }
+
+    /**
+     * Sets the health data
+     * @param {Object} dataObject The object to be stored as health data
+     */
+    function setHealthData(dataObject) {
+        if (!shouldLogHealthData()) {
+            return;
+        }
+        PreferencesManager.setViewState(HEALTH_DATA_STATE_KEY, dataObject);
+    }
+
+    /**
+     * Returns health data logged for the given key
+     * @returns {Object} Health Data object for the key or undefined if no health data stored
+     */
+    function getHealthDataLog(key) {
+        var healthData = getStoredHealthData();
+        return healthData[key];
+    }
+
+    /**
+     * Sets the health data for the given key
+     * @param {Object} dataObject The object to be stored as health data for the key
+     */
+    function setHealthDataLog(key, dataObject) {
+        var healthData = getStoredHealthData();
+        healthData[key] = dataObject;
+        setHealthData(healthData);
+    }
+
+    /**
+     * Clears all the health data recorded till now
+     */
+    function clearHealthData() {
+        PreferencesManager.setViewState(HEALTH_DATA_STATE_KEY, {});
+        //clear the performance related health data also
+        PerfUtils.clear();
+    }
+
+    /**
+     * Enable or disable health data logs
+     * @param {boolean} enabled true to enable health logs
+     */
+    function setHealthLogsEnabled(enabled) {
+        logHealthData = enabled;
+        if (!enabled) {
+            clearHealthData();
+        }
+    }
+
+    /**
+     * Whenever a file is opened call this function. The function will record the number of times
+     * the standard file types have been opened. We only log the standard filetypes
+     * @param {String} filePath          The path of the file to be registered
+     * @param {boolean} addedToWorkingSet set to true if extensions of files added to the
+     *                                    working set needs to be logged
+     */
+    function fileOpened(filePath, addedToWorkingSet) {
+        if (!shouldLogHealthData()) {
+            return;
+        }
+        var fileExtension = FileUtils.getFileExtension(filePath),
+            language = LanguageManager.getLanguageForPath(filePath),
+            healthData = getStoredHealthData(),
+            fileExtCountMap = [];
+        healthData.fileStats = healthData.fileStats || {
+            openedFileExt     : {},
+            workingSetFileExt : {}
+        };
+        if (language.getId() !== "unknown") {
```

Comment:
Belated, but: is there a reason we don't want to log unrecognized file extensions too?  Seems like it'd be a useful way of identifying gaps in our default set of syntax highlighters & file-extension mappings...